### ISSUE TITLE
numpy version issues on PyPI

### ISSUE THREAD

**Issue by bdice** [glotzerlab/libgetar#13]
This package is seeing the same numpy issues that affected `gsd` and `freud`. It needs to be rebuilt with an older version of NumPy for PyPI deployment.

Issues:
https://github.com/glotzerlab/gsd/issues/23
https://github.com/glotzerlab/freud/issues/243

Solutions:
https://github.com/glotzerlab/gsd/pull/24
https://github.com/glotzerlab/freud/pull/252

With numpy 1.15.4:
```python
>>> import gtar
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../lib/python3.7/site-packages/gtar/__init__.py", line 2, in <module>
    from . import _gtar
  File "__init__.pxd", line 872, in init gtar._gtar
ValueError: numpy.ufunc has the wrong size, try recompiling. Expected 192, got 216
```
After upgrading to numpy 1.16.3, it works fine.

**Comment by klarh** [comment]
The problem isn't one of rebuilding the package (as no binaries are currently distributed), but instead of under-specified dependencies and interactions with PEP 518. With build isolation enabled, `pip` will use the latest version of numpy in its isolated build environment, which will potentially clash with whatever has already been installed. It would be nice if we could specify to base the build environment off of the current python environment (and therefore use the current version if it's already installed), but at a glance I don't see an option for that.

Probably the simplest solution is to specify a `V` with `numpy<=V` as a build requirement and `numpy>=V` as an install requirement. We could also advise users to disable the build isolation in pip, but that seems like more trouble than it's worth.

**Comment by vyasr** [comment]
#252 addresses this issue for wheels on pip, which are now built against numpy 1.10. We specifically use build isolation and avoid pep517 to make that work. The correct solution for source builds is probably what you suggested, but we have a bit of a problem because source builds of NumPy don't compile unless you manually remove certain files (I believe it's the random number implementation). I don't quite remember the reason anymore, but I manually delete those files when building from source on CircleCI. I can try and find the earliest NumPy version that successfully installs from pip and set that as a minimum build requirement in the pyproject.toml, but it will be a slightly higher number than the install requirement set in setup.py. @klarh do you think that's sufficient?

**Comment by klarh** [comment]
Should be fixed as of v1.0.0, at least for most practical cases.
